internal/cmd/gocdk/launch.go

```diff
@@ -15,11 +15,170 @@
+	if err != nil {
+		return xerrors.Errorf("gocdk launch: %w", err)
+	}
+
+	// Read the launch specifier from the biome's Terraform output.
+	tfOutput, err := tfReadOutput(ctx,
+		findBiomeDir(moduleRoot, biome),
+		pctx.overrideEnv("TF_IN_AUTOMATION=1"))
+	if err != nil {
+		return xerrors.Errorf("gocdk launch: %w", err)
+	}
+
+	// Launch the application.
+	launchURL, err := launcher.Launch(ctx, &launchInput{
+		dockerImage: *imageName,
+		specifier:   launchSpecifier(tfOutput["launch_specifier"].mapValue()),
+		// TODO(light): Pass environment variables from tfOutput into env.
+	})
+	if err != nil {
+		return xerrors.Errorf("gocdk launch: %w", err)
+	}
+	fmt.Fprintf(pctx.stdout, "Serving at %s\n", launchURL)
+	return nil
+}
+
+// Launcher is the interface for any type that can launch a Docker image.
+type Launcher interface {
+	Launch(ctx context.Context, input *launchInput) (*url.URL, error)
+}
+
+// launchInput is the input to a launcher.
+type launchInput struct {
+	// dockerImage specifies the image name and tag of the local Docker image to
+	// deploy. If the local image does not exist, then the launcher should return
+	// an error.
+	dockerImage string
+
+	// env is the set of additional environment variables to set. It should not
+	// include PORT nor should it contain multiple entries for the same variable
+	// name.
+	env []string
+
+	// specifier is the set of arguments passed from a biome's Terraform module.
+	specifier launchSpecifier
+}
+
+// newLauncher creates the launcher for the given name.
+func newLauncher(ctx context.Context, pctx *processContext, launcherName string) (Launcher, error) {
+	logger := log.New(pctx.stderr, "gocdk: ", log.Ldate|log.Ltime)
+	switch launcherName {
+	case "local":
+		return &localLauncher{
+			logger:    logger,
+			dockerEnv: pctx.env,
+			dockerDir: pctx.workdir,
+		}, nil
+	default:
+		return nil, xerrors.Errorf("prepare launcher: unknown launcher %q", launcherName)
+	}
+}
+
+// localLauncher starts local Docker containers.
+type localLauncher struct {
+	logger    *log.Logger
+	dockerEnv []string
+	dockerDir string
+}
+
+// Launch implements Launcher.Launch.
+func (local *localLauncher) Launch(ctx context.Context, input *launchInput) (*url.URL, error) {
+	dockerArgs := []string{
+		"run",
+		"--rm",
+		"--detach",
+		"--publish", "8080:8080",
+	}
+	for _, v := range input.env {
+		dockerArgs = append(dockerArgs, "--env", v)
+	}
+	dockerArgs = append(dockerArgs, "--env", "PORT=8080")
+	dockerArgs = append(dockerArgs, input.dockerImage)
+
+	c := exec.CommandContext(ctx, "docker", dockerArgs...)
+	c.Env = local.dockerEnv
+	c.Dir = local.dockerDir
+	out, err := c.CombinedOutput()
+	if err != nil {
+		if len(out) == 0 {
+			return nil, xerrors.Errorf("local launch: docker run: %w", err)
+		}
+		return nil, xerrors.Errorf("local launch: docker run:\n%s", out)
+	}
+
+	local.logger.Printf("Docker container %s started, waiting for healthy...", bytes.TrimSuffix(out, []byte("\n")))
```

Comment:
How do they stop it?

Reply:
Is `Launch` going to support an `Unlaunch` or `Stop` command? That would seem reasonable to me.

Reply:
(maybe file an issue for that).

And consider just extending this `Printf` with a note on how to stop the container.

Reply:
Added log statement. Unlaunch/stop is likely going to be too provider-specific since you need to give a handle to a previous deployment. Honestly, I think the more common situation will be that folks just run `gocdk launch -image=MYIMAGE:OLDTAG` to do a rollback.